### PR TITLE
solves #303

### PR DESCRIPTION
**Pull request by wxyz-abcd**

### CODE REVIEW

**Comment by murat-dogan** [comment]
Thank you for the PR.

I am mostly OK with it.
There are lots of eslint errors;
```
/home/murat/js/fork/node-datachannel/test/jest-tests/polyfill.test.ts
   31:4  warning  Missing return type on function                        @typescript-eslint/explicit-function-return-type
   32:9  error    'binaryData' is never reassigned. Use 'const' instead  prefer-const
   33:9  error    'dv' is never reassigned. Use 'const' instead          prefer-const
   40:4  warning  Missing return type on function                        @typescript-eslint/explicit-function-return-type
   41:9  error    'dv' is never reassigned. Use 'const' instead          prefer-const
   49:4  error    Unexpected var, use let or const instead               no-var
   50:4  warning  Missing return type on function                        @typescript-eslint/explicit-function-return-type
   61:4  warning  Missing return type on function                        @typescript-eslint/explicit-function-return-type
   62:5  error    Unexpected var, use let or const instead               no-var
  129:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  130:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  131:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  132:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  133:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  134:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  135:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  136:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  137:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  138:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  139:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  140:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  141:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  142:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  143:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  144:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  145:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  146:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  147:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
  148:6  error    Avoid calling `expect` conditionally`                  jest/no-conditional-expect
```

And the code is a little bit hard to follow. Maybe comments will help.

**Comment by wxyz-abcd** [comment]
OK, fixed them all hopefully, and added comments too. Hope it is enough.

**Comment by murat-dogan** [comment]
Thank you.
